cartoframes/data/observatory/catalog/entity.py

```diff
@@ -142,7 +144,12 @@ def _download(self, credentials, file_path=None, limit=None, order_by=None, sql_
                 log.info(_GEOGRAPHY_READ_MSG.format(file_path))
         else:
             dataframe = pd.read_csv(rows)
-            return dataframe
+            gdf = GeoDataFrame(dataframe)
+
+            if has_geometry(dataframe):
```

Comment:
The function `has_geometry` check the geometry of an existing GeoDataFrame, not for a DataFrames, so this won't work. We should implement the following logic instead:
- Check if the existing dataframe contains a raw geom column if:
  - The entity is a Geography or
  - The entity is a Dataset and add_geom is True
- In every case check that the column is there and can be decoded:
  - decode_geometry(df['geom'])
- If it works, create the GeoDataFrame with that geometry
- PD: we need to decide if we should keep the original geom column or not